table-cat/table-cat.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python
+
+"""
+Script to concatenate multiple files (table-format) with common header,
+also works for gzip files, with output in non-gzip table format.
+"""
+import sys
+import argparse
+import gzip
+
+parser = argparse.ArgumentParser()
+parser.add_argument("-i", "--tables", nargs='*',
+                    help="Input multiple table with same headers")
+parser.add_argument("-z", "--gzipped", action='store_true',
+                    help="Use this option for zipped files")
+args = parser.parse_args()
+
+#Get headers
+headerlines = []
+if args.gzipped:
+    for file in args.tables:
+        with gzip.open(file, 'rb') as infile:
+            first_line = infile.readline().decode('ascii')
+            headerlines += [first_line.strip()]
+else:
+    for file in args.tables:
+        with open(file) as infile:
+            first_line = infile.readline()
+            headerlines += [first_line.strip()]
+
+if all(headers == headerlines[0] for headers in headerlines):
+    print(headerlines[0])
+    if args.gzipped:
+        for file in args.tables:
+            with gzip.open(file, 'rb') as infile:
+                next(infile)
+                for line in infile:
+                    print(line.decode('ascii').strip())
+    else:
+        for file in args.tables:
+            with open(file) as infile:
+                next(infile)
+                for line in infile:
+                    print(line.strip())
```

Comment:
Suggestion: This is essentially code repetition (code smell). The only thing that changes is the file opening command. Could have used a function like `def open_filehandle(file, gunzip)` that opens the file and returns it. Then you could use this function instead of the `with ...` and merge the 2 if..for statements.

Reply:
```python
for file in args.tables:
    infile = open_filehandle(file, args.gunzip)
    next(infile)
    for line in infile:
         print(line.strip())
```